src/src.js

```diff
@@ -43,19 +45,154 @@ class Game {
     }
 
     initGame() {
-        this.spaces = []
-        this.generateSpace()
-        this.gameOver = false
+        this.spaces = [];
+        this.generateSpace();
+        this.gameOver = false;
+        this.populateMap();
     }
 
+
+
+
     generateSpace() {
         this.addSpace(new Space())
     }
 
     addSpace(space) {
         this.spaces.push(space)
     }
+
+
+    generateMap(col, row) {
+
+        // create array of size x, y
+        // populate with values of y
+
+        // should i be defining all these variables as this.map etc??
+
+        let map = [];
```

Comment:
in js it's preferable to use "const" instead of "let" unless you really need to reuse a variable (such as in a loop). Arrays and Objects can be declare as "const" and then effected so long as you aren't reusing them.

ie: 

const arr = []
arr.push('foo') // good'
arr = {} // bad, need to declare arr with 'let'

const foo = 'bar' // good
foo = 'baz' // bad, need to have declared original with 'let'

Reply:
I think it's preferable mostly for memory management/performance considerations. But also helps reduce bugs over time :)

Reply:
cool, makes sense!